packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -548,6 +398,16 @@ contract Bridge is EssentialContract, IBridge {
         return _msgHash ^ bytes32(uint256(Status.FAILED));
     }
 
+    /// @notice Returns the minimal gas limit required for sending a given message.
+    /// @param _message The message.
+    /// @return The minimal gas limit required for sending this message.
+    function getMessageMinGasLimit(Message calldata _message) public pure returns (uint32) {
+        unchecked {
+            uint256 calldataCost = (_message.data.length + 192) >> 4;
+            return uint32(GAS_RESERVE + calldataCost + 1);
```

Comment:
I think this effectively reserves `GAS_RESERVE` of the gas limit mainly for the proof verification. So two things can happen:
- `GAS_RESERVE` < actual proof verification cost: the relayer runs the risk to lose money because gas used > gas limit, so depending on the block basefee, the fee paid by the user may be capped and the relayer will maye have spent more money processing the bridge message than it got because the call invocation has a gas limit that allows more gas to be used than the max gas limit set by the user.
- `GAS_RESERVE` > actual proof verification cost: this gas got reserved for the proof verification but will be refunded to the user because not usable. Only downside I guess is that the user has to send more ETH to the destination chain than necessary (but that will always be the case, but maybe a bit worse than required depending on this value).

Compared to just using whatever gas is left seems to be that the gas limit for the invoked call is deterministic and not gameable by the relayer. So in this way they can be seen as separate steps each with its own fee payment, but with a single shared fee pool by shifting some of the risk/complexity of profitability to the relayer, and depending on how smart the relayer is it could also be beneficial to the user.

I think it works.

Reply:
I think we need to modify this contract later after gathering some data to fine-tune some constants.